src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IVariableDeclaration.cs

```diff
@@ -0,0 +1,297 @@
+        public void UsingStatementDeclaration()
+        {
+            string source = @"
+using System;
+
+class Program : IDisposable
+{
+    static void Main(string[] args)
+    {
+        using (/*<bind>*/Program p1 = new Program()/*</bind>*/)
+        {
+        }
+    }
+
+    public void Dispose() {}
+}
+";
+            string expectedOperationTree = @"
+IUsingStatement (OperationKind.UsingStatement)
+  IVariableDeclarationStatement (1 variables) (OperationKind.VariableDeclarationStatement)
+    IVariableDeclaration: Program p1 (OperationKind.VariableDeclaration)
+      Initializer: IObjectCreationExpression (Constructor: Program..ctor()) (OperationKind.ObjectCreationExpression, Type: Program)
+  IBlockStatement (0 statements) (OperationKind.BlockStatement)
+";
+            VerifyOperationTreeForTest<VariableDeclarationSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17599, "https://github.com/dotnet/roslyn/issues/17599")]
+        public void UsingStatementMultipleDeclarations()
+        {
+            string source = @"
+using System;
+
+class Program : IDisposable
+{
+    static void Main(string[] args)
+    {
+        using (/*<bind>*/Program p1 = new Program(), p2 = new Program()/*</bind>*/)
+        {
+        }
+    }
+
+    public void Dispose() {}
+}
+";
+            string expectedOperationTree = @"
+IUsingStatement (OperationKind.UsingStatement)
+  IVariableDeclarationStatement (2 variables) (OperationKind.VariableDeclarationStatement)
+    IVariableDeclaration: Program p1 (OperationKind.VariableDeclaration)
+      Initializer: IObjectCreationExpression (Constructor: Program..ctor()) (OperationKind.ObjectCreationExpression, Type: Program)
+    IVariableDeclaration: Program p2 (OperationKind.VariableDeclaration)
+      Initializer: IObjectCreationExpression (Constructor: Program..ctor()) (OperationKind.ObjectCreationExpression, Type: Program)
+  IBlockStatement (0 statements) (OperationKind.BlockStatement)
+";
+            VerifyOperationTreeForTest<VariableDeclarationSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17599, "https://github.com/dotnet/roslyn/issues/17599")]
+        public void ForLoopDeclaration()
+        {
+            string source = @"
+using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        for (/*<bind>*/int i = 0/*</bind>*/; i < 0; i++)
+        {
+
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+IForLoopStatement (LoopKind.For) (OperationKind.LoopStatement)
+  Condition: IBinaryOperatorExpression (BinaryOperationKind.IntegerLessThan) (OperationKind.BinaryOperatorExpression, Type: System.Boolean)
+      Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+      Right: ILiteralExpression (Text: 0) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0)
+  Local_1: System.Int32 i
```

Comment:
Minor: I think we can adjust the dumper to first emit the locals declared, then the Before part, then Condition, and finally AtLoopBottom. http://source.roslyn.io/#Roslyn.Test.Utilities/Compilation/OperationTreeVerifier.cs,303